comp/core/workloadmeta/workloadmeta.go

```diff
@@ -138,12 +142,70 @@ func newWorkloadMetaOptional(deps dependencies) optionalProvider {
 	c := newWorkloadMeta(deps)
 
 	return optionalProvider{
-		Comp:          optional.NewOption(c.Comp),
-		FlareProvider: c.FlareProvider,
-		Endpoint:      c.Endpoint,
+		Comp:                         optional.NewOption(c.Comp),
+		FlareProvider:                c.FlareProvider,
+		WorkloadListEndpoint:         c.WorkloadListEndpoint,
+		PodContainerMetadataEndpoint: c.PodContainerMetadataEndpoint,
 	}
 }
 
+type PodContainerMetadata struct {
+	Name       string                `json:"name"`
+	Cmd        []string              `json:"cmd"`
+	Entrypoint []string              `json:"entrypoint"`
+	Image      OrchestratorContainer `json:"image"`
```

Comment:
Are you sure about the type here? [OrchestratorContainer is a reference to a Container with
orchestrator-specific data attached to it.](https://github.com/DataDog/datadog-agent/blob/7debf30a42aaffddafffaa3f619469c47b98b1e6/comp/core/workloadmeta/types.go#L442C3-L443C46)